sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/LanguageDetection.java

```diff
@@ -122,6 +122,9 @@ private static String getDetails(Language detectedLanguage, PathPattern[] patter
   }
 
   static String sanitizeExtension(String suffix) {
-    return StringUtils.lowerCase(StringUtils.removeStart(suffix, "."));
+    if (!suffix.contains(".")) {
+      return "." + StringUtils.lowerCase(suffix);
```

Comment:
What would be the behaviour if customer passes 

1.  .field-meta.xml
2.  field-meta.xml [ prefix (.) will be absent in this case]

Reply:
@anurag-ar 
1. It will match all files ending with **.field-meta.xml**
2. Same as above, **field-meta.xml** will be matched.

I am only modifying suffix when there is no . present. For e.g if customer sends just "xml" it will get converted to ".xml" because absence of dot is assumed to be an extension.